framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1208,9 +1208,14 @@ public boolean expunge(final ID id) {
         }
     }
 
-    // FIXME: Does not work for joins.
     @Override
     public int expunge(final SearchCriteria<T> sc) {
+        return expunge(sc, -1);
+    }
+
+    // FIXME: Does not work for joins.
+    @Override
+    public int expunge(final SearchCriteria<T> sc, long limit) {
```

Comment:
```suggestion
    public int expunge(final SearchCriteria<T> sc, final Filter filter) {
```
I would suggest keeping the parameters for this method similar to how it is for other methods for this class.

Reply:
I'd rather stick with the simpler long limit, as I don't see any need to use the Filter's features on an expunge.

Reply:
Added explanation in the comment below.
https://github.com/apache/cloudstack/pull/8740#discussion_r1513377957